Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -62,18 +59,14 @@ public class PeptideWithSetModifications : ProteolyticPeptide
             }
 
             FullSequence = sequence;
-            GetModsAfterDeserialization(allKnownMods, out _baseSequence);
+            GetModsAfterDeserialization(allKnownMods);
```

Comment:
I believe the `out _baseSequence` part is so the base sequence gets set if the protein is not known (for example if you construct a peptide with the `PeptideWithSetModifications(string sequence ...)` constructor). can you check that you can serialize/deserialize these types of peptides and that their base sequences are set correctly?

Reply:
Yep, that's what was causing the unit test failures.